src/ffi.rs

```diff
@@ -221,7 +284,22 @@ fn test_gethostbyname2_r() {
         "localhost.".to_string(),
         libc::AF_INET6,
     );
-
     let res = result.expect("Should resolve IPv6 localhost.");
     println!("{:?}", res);
 }
+
+#[test]
+fn test_gethostbyaddr_r() {
+    disable_internal_nscd();
+
+    let v4test = LibcIp::V4([127,0,0,1]);
+    let result: Result<Hostent, anyhow::Error> = gethostbyaddr_r(v4test);
```

Comment:
You don't need to specify the type of the result, especially if you're expecting and unwrapping the error types away.

This can probably just be `let hostent = gethostbyaddr_r(v4test).unwrap();` (and maybe even inline the function arg)